rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/model/ParametricSummary.java

```diff
@@ -227,32 +227,32 @@ private CompletableFuture<Versioned<ParametricSummary>> createSummary(
     }
 
     public class MessagesOnlyScheduledSummary extends NullSummary {
-        private final InterruptibleFuture<List<Diagnostic>> messages;
+        private final InterruptibleFuture<Lazy<List<Diagnostic>>> messages;
```

Comment:
ah, my original comment was wrong, we were already storing the future, I thought extractMessages was called everytime. so my bad, I don't think we need lazy here.